txs/cache_test.go

```diff
@@ -1021,41 +1069,37 @@ func TestCache_LinkTXsWithProposal(t *testing.T) {
 	lid0 := types.NewLayerID(97)
 	pid0 := types.ProposalID{1, 2, 3}
 	// take the first tx out of each account for proposal 0
-	txs0 := make([]*types.MeshTransaction, 0, len(mtxsByAccount))
 	tids0 := make([]types.TransactionID, 0, len(mtxsByAccount))
 	totalNumTXs := 0
 	for _, mtxs := range mtxsByAccount {
 		totalNumTXs += len(mtxs)
-		txs0 = append(txs0, mtxs[0])
 		tids0 = append(tids0, mtxs[0].ID)
-		mtxs[0].LayerID = lid0
 	}
 	require.NoError(t, tc.LinkTXsWithProposal(tc.db, lid0, pid0, tids0))
-	for _, mtxs := range mtxsByAccount {
-		checkTX(t, tc.cache, mtxs[0])
-	}
-	checkTXStateFromDB(t, tc.db, txs0, types.MEMPOOL)
-
+	// take the second tx out of each account for proposal 1
 	lid1 := lid0.Add(1)
 	pid1 := types.ProposalID{2, 3, 4}
-	// take the second tx out of each account for proposal 1
-	txs1 := make([]*types.MeshTransaction, 0, len(mtxsByAccount))
 	tids1 := make([]types.TransactionID, 0, len(mtxsByAccount))
 	for _, mtxs := range mtxsByAccount {
 		if len(mtxs) > 1 {
-			txs1 = append(txs1, mtxs[1])
 			tids1 = append(tids1, mtxs[1].ID)
-			mtxs[1].LayerID = lid1
 		}
 	}
 	require.NoError(t, tc.LinkTXsWithProposal(tc.db, lid1, pid1, tids1))
+
 	for _, mtxs := range mtxsByAccount {
-		if len(mtxs) > 1 {
-			checkTX(t, tc.cache, mtxs[1])
+		checkTX(t, tc.cache, mtxs[0].ID, lid0, types.EmptyBlockID)
+		cnt := len(mtxs)
+		if cnt > 1 {
+			checkTX(t, tc.cache, mtxs[1].ID, lid1, types.EmptyBlockID)
+		} else if cnt > 2 {
+			for _, mtx := range mtxs[2:] {
+				checkTX(t, tc.cache, mtx.ID, types.LayerID{}, types.EmptyBlockID)
+			}
```

Comment:
The `else if cnt > 2` branch will never be taken.

Reply:
why? each account has N txs in the pool, N is a random number bwtn 1 and 10.

Reply:
first branch is always true if second branch is true, so it will always go through the first

Reply:
ahhh... my bad. thank you! fixed